corehq/apps/app_manager/translations.py

```diff
@@ -545,26 +557,26 @@ def update_form_translations(sheet, rows, missing_cols, app):
                             new_translation = fallback
                             break
 
-                if new_translation:
-                    # Create the node if it does not already exist
-                    if not value_node.exists():
-                        e = etree.Element(
-                            "{f}value".format(**namespaces), attributes
-                        )
-                        text_node.xml.append(e)
-                        value_node = WrappedNode(e)
-                    # Update the translation
-                    value_node.xml.tail = ''
-                    for node in value_node.findall("./*"):
-                        node.xml.getparent().remove(node.xml)
-                    escaped_trans = escape_output_value(new_translation)
-                    value_node.xml.text = escaped_trans.text
-                    for n in escaped_trans.getchildren():
-                        value_node.xml.append(n)
+                if trans_type == 'default':
+                    value_node = next(
+                        n for n in text_node.findall("./{f}value")
+                        if 'form' not in n.attrib
+                    )
+                    markdown_node = text_node.find("./{f}value[@form='markdown']")
+                    has_markdown = _looks_like_markdown(new_translation)
+                    had_markdown = markdown_node.exists()
+                    vetoed_markdown = not had_markdown and _looks_like_markdown(etree.tostring(value_node.xml))
```

Comment:
Oh actually it shouldn't affect this, since veto markdown says it doesn't have markdown if markdown isn't written, and if vellum:ignore="markdown" is in the form then those markdown nodes should not be written.